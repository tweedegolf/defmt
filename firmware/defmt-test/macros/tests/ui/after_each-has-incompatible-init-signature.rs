fn main() {}

#[defmt_test_macros::tests]
mod tests {
    #[init]
    fn init() -> u32 {
        0_u32
    }

    #[after_each]
    fn say(value: &mut u16) {
        assert!(true);
    }
}
