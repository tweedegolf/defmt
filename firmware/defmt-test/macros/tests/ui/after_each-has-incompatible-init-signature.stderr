error: this type must match `#[init]`s return type: u32
  --> tests/ui/after_each-has-incompatible-init-signature.rs:11:24
   |
11 |     fn say(value: &mut u16) {
   |                        ^^^
